kali_cloud_init.tf

```diff
@@ -121,4 +121,64 @@ data "cloudinit_config" "kali_cloud_init_tasks" {
       merge_type   = "list(append)+dict(recurse_array)+str()"
     }
   }
+
+  # Create a credentials file for the VNC user that can be used to configure
+  # the AWS CLI to write to the assessment artifact export S3 bucket.
+  # For details, see
+  # https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#using-a-configuration-file
+  #
+  # Input variables are:
+  # * aws_access_key_id - the AWS access key ID
+  # * aws_region - the AWS region of the S3 bucket
+  # * aws_secret_access_key - the AWS secret access key
+  # * permissions - the octal permissions to assign the AWS configuration
+  # * vnc_username - the username associated with the VNC user
+  dynamic "part" {
+    # Only include this block if var.assessment_artifact_export_enabled is true.
+    for_each = var.assessment_artifact_export_enabled ? [0] : []
```

Comment:
We're using `for_each` because it's a `dynamic` block right?

Reply:
Correct.  I'm following the same pattern that we established in #207:
https://github.com/cisagov/cool-assessment-terraform/blob/840cbc636c801e9812630eb08c75c89799d00b4d/kali_cloud_init.tf#L78-L81